File: frontend/src/types/report.ts
export type WeatherCondition =
  | "sunny"
  | "cloudy"
  | "drizzle"
  | "heavy_rain"
  | "flood";

export interface Report {
  id: string;
  condition: WeatherCondition;
  lat: number;
  lng: number;
  is_onsite: boolean;
  created_at: string;
  expires_at: string;
}

export interface CreateReportPayload {
  condition: WeatherCondition;
  lat: number;
  lng: number;
  is_onsite: boolean;
}

export interface LocationRequest {
  id: string;
  lat: number;
  lng: number;
  area_name: string;
  message: string;
  fulfilled_count: number;
  created_at: string;
  expires_at: string;
}

export interface CreateRequestPayload {
  lat: number;
  lng: number;
  area_name: string;
  message: string;
}

File: backend/models/report.go
package models

import "time"

type Report struct {
    ID        string    `db:"id" json:"id"`
    Condition string    `db:"condition" json:"condition"`
    Lat       float64   `db:"lat" json:"lat"`
    Lng       float64   `db:"lng" json:"lng"`
    IsOnsite  bool      `db:"is_onsite" json:"is_onsite"`
    CreatedAt time.Time `db:"created_at" json:"created_at"`
    ExpiresAt time.Time `db:"expires_at" json:"expires_at"`
}

type CreateReportRequest struct {
    Condition string  `json:"condition"`
    Lat       float64 `json:"lat"`
    Lng       float64 `json:"lng"`
    IsOnsite  bool    `json:"is_onsite"`
}
